dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/S3Utils.java

```diff
@@ -175,21 +175,36 @@ public String getFileName(ResourceType resourceType, String tenantCode, String f
     }
 
     @Override
-    public void download(String tenantCode, String srcFilePath, String dstFile, boolean deleteSource, boolean overwrite) throws IOException {
+    public void download(String tenantCode, String srcFilePath, String dstFilePath, boolean deleteSource, boolean overwrite) throws IOException {
+        File dstFile = new File(dstFilePath);
+        File dstFileParentDirectory = dstFile.getParentFile();
+        if (dstFile.exists()) {
+            if (dstFile.isDirectory()) {
+                logger.error("destination file must be a file, but {} is a folder", dstFilePath);
+                throw new IOException("destination file must be a file, but " + dstFilePath +" is a folder");
+            } else if (!overwrite) {
+                logger.info("the destination file {} already exists, download operation will be ignored", dstFilePath);
+                return;
+            }
+        } else {
+            if (!dstFileParentDirectory.mkdirs() && !dstFileParentDirectory.exists()) {
+                throw new IOException("failed to create parent directory of destination file, directory is " + dstFileParentDirectory.getAbsolutePath());
+            }
```

Comment:
Can we directly use `org.apache.commons.io.Files.mkdirs(dstFileParentDirectory)`
```suggestion
            FileUtils.mkdirs(dstFileParentDirectory)
            }
```

Reply:
* If the failed to create dir exception is not thrown in time, it will unnecessarily request s3 resources
* I will replaced it with FileUtils on next push

Reply:
Ok, `mkdirs` will throw exception if createDir failed.

Reply:
> Can we directly use `org.apache.commons.io.Files.mkdirs(dstFileParentDirectory)`

I have updated this part of the code